### PR TITLE
GG-28760 Deprecate DataRegionMetrics#OffheapUsedSize, add TotalUsedSize

### PR DESCRIPTION
**Pull request by ibessonov**

### CODE REVIEW

File: modules/platforms/dotnet/Apache.Ignite.Core/IDataRegionMetrics.cs [hunk 1]
```diff
@@ -43,6 +43,11 @@ public interface IDataRegionMetrics
         /// </summary>
         long TotalAllocatedSize { get; }
 
+        /// <summary>
+        /// Gets a total number of pages used for storing the data.
```
Comment (ptupitsyn) [inline modules/platforms/dotnet/Apache.Ignite.Core/IDataRegionMetrics.cs:47]: Please put here the same detailed comment that we have in `DataRegionMetrics.java`

**Review by ptupitsyn — CHANGES_REQUESTED** [review]
Can we add tests for the new property? Ideally both in Java and C#.

File: modules/platforms/dotnet/Apache.Ignite.Core/IDataRegionMetrics.cs (hunk 1 above)
Comment (ibessonov) [reply modules/platforms/dotnet/Apache.Ignite.Core/IDataRegionMetrics.cs]: Done

**Review by ptupitsyn — APPROVED** [review]
I've added a test to C# part, now we are good to go

**Pull request closed without merging**